cmd/juju/status/output_tabular.go

```diff
@@ -97,95 +97,98 @@ func FormatTabular(writer io.Writer, forceColor bool, value interface{}) error {
 		tw.Flush()
 	}
 
-	units := make(map[string]unitStatus)
-	outputHeaders("App", "Version", "Status", "Scale", "Charm", "Store", "Rev", "OS", "Notes")
-	tw.SetColumnAlignRight(3)
-	tw.SetColumnAlignRight(6)
-	for _, appName := range naturalsort.Sort(stringKeysFromMap(fs.Applications)) {
-		app := fs.Applications[appName]
-		version := app.Version
-		// Don't let a long version push out the version column.
-		if len(version) > maxVersionWidth {
-			version = version[:truncatedWidth] + ellipsis
-		}
-		// Notes may well contain other things later.
-		notes := ""
-		if app.Exposed {
-			notes = "exposed"
-		}
-		w.Print(appName, version)
-		w.PrintStatus(app.StatusInfo.Current)
-		scale, warn := fs.applicationScale(appName)
-		if warn {
-			w.PrintColor(output.WarningHighlight, scale)
-		} else {
-			w.Print(scale)
+	if len(fs.Applications) > 0 {
```

Comment:
I wonder if these stanzas could all be extracted to helper funcs to reduce the huge indented blocks
func printApplications(apps) {
if len(apps) == 0 {
return
}
print....
}

then status could be
printMachines(machines)
printApplications(apps)
etc